plan/relations.go

```diff
@@ -741,6 +776,12 @@ func (j *JoinRel) CopyWithExpressionRewrite(rewriteFunc RewriteFunc, newInputs .
 	return &join, nil
 }
 
+func (j *JoinRel) ChangeMapping(mapping []int32) error {
+	newMapping, err := ChangeMapping(j, mapping)
+	j.mapping = newMapping
```

Comment:
If there is an error, we lose the old mapping if I reading this right. If err, I would expect no change.

Reply:
Fixed the bug in the ChangeMapping helper and added a test.